docs/reference/contributing/target/sai.md

```diff
@@ -0,0 +1,100 @@
+
+This highly configurable interface has a wide range of elements you can adjust:
+
+- Word length.
+- Data length (inside words).
+- Clocks polarity and phase.
+- Data alignment.
+
+Please refer to your device's reference manual for more details on its capabilities.
+
+This API provides a way to receive or send audio sample through this interface.
+
+Transfer and free functions, plus an init function, comprise this API.
+
+`sai_init` is in charge of initializing this the whole interface and ensuring that the device supports the requested format. It also has to make sure the clocks are configured to generate the requested frequency with a reasonable error margin (depending on the master/slave tolerance).
+
+<span class="warnings">**Warning:** We are introducing the SAI API in an upcoming release of Mbed OS. This page documents code that exists on a feature branch of Mbed OS. You can find details on how it may affect you in the [implementing the SAI API](#implementing-the-sai-api) section.
+
+### Assumptions
+
+#### Defined behavior
+
+- `sai_init()` returns `SAI_RESULT_INVALID_PARAM` if at least one of the given parameters is undefined (NULL).
+- `sai_init()` returns `SAI_RESULT_ALREADY_INITIALIZED` if SAI is already initialized.
+- `sai_init()` returns `SAI_RESULT_CONFIG_UNSUPPORTED` if the device can never support this configuration.
+- `sai_init()` returns `SAI_RESULT_CONFIG_MISMATCH` if the device is not able to support this configuration now because of other 'live' constraints, such as a shared format or clock configuration with a sibling.
+- `sai_free()` does nothing if passed a NULL pointer.
+- `sai_free()` deinitializes and unclocks the unused part of the device.
+- You can reinitialize a device or block by using `sai_init()` after you use `sai_free()` on it.
+
+If the device is a *receiver*:
+
+- `sai_transfer()` returns false if the `sai_t` object is NULL.
+- `sai_transfer()` returns false if there's no sample in the FiFo.
+- `sai_transfer()` if `psample` is NULL; it pops 1 sample from the FiFo and returns true.
+- `sai_transfer()` if `psample` is not NULL; it pops 1 sample from the FiFo, stores it to the address pointed by `psample` and returns true.
+
+If the device is a *transmitter*:
+
+- `sai_transfer()` returns false if the `sai_t` object is NULL.
+- `sai_transfer()` returns false if the FiFo is full and you can't push `*psample`.
+- `sai_transfer()` if `psample` is NULL; it pushes one '0' sample to the FiFo and returns true.
+- `sai_transfer()` if `psample` is not NULL; it pushes the pointed sample to the FiFo and returns true.
+
+#### Undefined behavior
+
+- Calling any function other than `sai_init()` before the initialization of the SAI.
+- Calling any function other than `sai_init()` after calling `sai_free()`.
+
+#### Other requirements
+
+A target must also define these elements to allow tests to be run:
+
+- `#define SAI_DEFAULT_SAMPLE_RATE (xxxxxU)`.
```

Comment:
Can we describe what actually does it mean and why it's important?

Reply:
Is this new text ok ?